File: DPCodes/528-maxIncSubarraySum.py


def maxIncSubArraySum(arr):
    n=len(arr)
    ans=0
    cur=arr[0]
    for i in range(1,n):
        if arr[i]>arr[i-1]:
            cur+=arr[i]

        else:
            cur=arr[i]
        ans=max(ans,cur)
    return ans
arr=[10,20,30,5,10,50]
print(maxIncSubArraySum(arr))

File: greedy/528-maxIncSubarraySum.cpp
#include<iostream>
#include<vector>
using std::cout;
using VEC=std::vector<int>;

int maxIncreasingSubArraySum(const VEC& nums)
{
    if(nums.empty())
        return 0;
    int n=nums.size();

    int cur=nums[0],ans=nums[0];

    for(int i=1;i<n;++i)
    {
        if(nums[i]>nums[i-1])
            cur+=nums[i];
        else
            cur=nums[i];
        ans=std::max(ans,cur);
    }
    return ans;
}

int main()
{
    VEC arr={10,20,30,5,10,50};
    cout<<maxIncreasingSubArraySum(arr);
}